appserver/deployment/dol/src/main/java/com/sun/enterprise/deployment/BundleDescriptor.java

```diff
@@ -388,26 +385,24 @@ public void addRole(SecurityRoleDescriptor descriptor) {
     /**
      * Removes a role object from me.
      */
+    @Override
     public void removeRole(Role role) {
         this.getRoles().remove(role);
     }
 
     /**
      * Utility method for iterating the set of named descriptors in the supplied nameEnvironment
      */
-    protected Collection getNamedDescriptorsFrom(JndiNameEnvironment nameEnvironment) {
-        Collection namedDescriptors = new Vector();
-        for (Iterator itr = nameEnvironment.getResourceReferenceDescriptors().iterator(); itr.hasNext();) {
-            ResourceReferenceDescriptor resourceReference = (ResourceReferenceDescriptor) itr.next();
+    protected Collection<NamedDescriptor> getNamedDescriptorsFrom(JndiNameEnvironment nameEnvironment) {
+        Collection<NamedDescriptor> namedDescriptors = new Vector<>();
+        for (ResourceReferenceDescriptor resourceReference : nameEnvironment.getResourceReferenceDescriptors()) {
```

Comment:
This isn't a class variable to it's less important somewhat, but this still isn't thread safe. Whatever we do, I'd update the method so that each of the loops here uses the same convention